extension/content_script.js

```diff
@@ -203,23 +204,32 @@ function moveOne(dir) { // Sets start and end
 
 	timeTrackerView.updateTimer(tracker.getSentenceId());
 	highlight(tracker.getSentenceId());
- 
-	if (dir == direction.BACKWARD) {
-		scrollUp();
-	} else if (dir == direction.FORWARD) {
-		scrollDown();
-	}
+	scrollToTracker();
 	return true;
 }
 
-// Scroll up when tracker is above page
-function scrollUp() {
-	let verticalMargin = 200;
+// Manual movement is limited to once 200ms, or else if user taps arrow keys really quickly,
+// or presses down on them, there will be
+// too many UI events (e.g. highlighting, etc.) happening at once, making things very slow.
+// Debounce note: Will execute only after this function is uncalled for that amount of time.
+// The inactivity timer gets reset when function gets called while it is 'recovering'.
+let moveOneDebounced = _.debounce(moveOne, 200);
```

Comment:
only the manual movement gets this debouncing behavior.
currently keypress actually triggers lots of event calls in 1 go.

We didn't notice this because manual arrow-key-based movement used to rely on autoplay, which already has rate-throttling logic

Reply:
Good grief this is good

Reply:
Yeah TIL about this too LOL
And we're lucky we got the lodash library (_.fun()) cos the sentence tokenizer library needed it and so i ported it before hah.